rasa/nlu/featurizers/count_vectors_featurizer.py

```diff
@@ -220,51 +245,117 @@ def train(
-            self.vectorizer = None
-            return
+        if self.use_shared_vocab:
+            shared_vectorizer = create_vectorizer(
+                self.token_pattern,
+                self.strip_accents,
+                self.lowercase,
+                self.stop_words,
+                (self.min_ngram, self.max_ngram),
+                self.max_df,
+                self.min_df,
+                self.max_features,
+                self.analyzer,
+            )
+            self.vectorizer = {
+                attribute: shared_vectorizer for attribute in MESSAGE_ATTRIBUTES
+            }
+        else:
+            self.vectorizer = {
+                attribute: create_vectorizer(
+                    self.token_pattern,
+                    self.strip_accents,
+                    self.lowercase,
+                    self.stop_words,
+                    (self.min_ngram, self.max_ngram),
+                    self.max_df,
+                    self.min_df,
+                    self.max_features,
+                    self.analyzer,
+                )
+                for attribute in MESSAGE_ATTRIBUTES
+            }
+
+        cleaned_attribute_texts = {}
+
+        for attribute in MESSAGE_ATTRIBUTES:
+            lem = [
+                self._get_message_text_by_attribute(example, attribute)
+                for example in training_data.intent_examples
+            ]
+            self._check_OOV_present(lem)
+            cleaned_attribute_texts[attribute] = lem
+
+        combined_cleaned_texts = []
+        if self.use_shared_vocab:
+            for attribute in MESSAGE_ATTRIBUTES:
+                combined_cleaned_texts += cleaned_attribute_texts[attribute]
+
+        featurized_attributes = {}
+        # noinspection PyPep8Naming
+
+        for attribute in MESSAGE_ATTRIBUTES:
+
+            try:
+                if self.use_shared_vocab:
+                    self.vectorizer[attribute].fit(combined_cleaned_texts)
+                else:
+                    self.vectorizer[attribute].fit(cleaned_attribute_texts[attribute])
+            except ValueError as e:
+                logger.warning(
+                    "Unable to train CountVectorizer for message attribute {0}. "
+                    "Returning with untrained CountVectorizer".format(attribute)
+                )
+                continue
+
+            self.is_trained[attribute] = True
+
+            featurized_attributes[attribute] = (
+                self.vectorizer[attribute]
+                .transform(cleaned_attribute_texts[attribute])
+                .toarray()
+            )
 
         for i, example in enumerate(training_data.intent_examples):
             # create bag for each example
-            example.set(
-                "text_features",
-                self._combine_with_existing_text_features(example, X[i]),
-            )
+
+            for attribute in MESSAGE_ATTRIBUTES:
+                if len(cleaned_attribute_texts[attribute][i]):
```

Comment:
@tabergma This is to check if the text for that attribute actually had anything or was None. I can't use `.get()` over the `example` object because it won't work for `text` attribute.
I'll add this as a comment in the code.

Reply:
Ok. Can you then please also use
```
if cleaned_attribute_texts[attribute][i]:
```
The `len()` was a bit confusing.

Reply:
Sure